### PR TITLE
Docs apis clients restructure

### PR DESCRIPTION
**Pull request by christinaausley**
## What is the purpose of the change

**Paired work with @pepopowitz 🥳 in building out React card component!**

Resolves https://github.com/camunda/developer-experience/issues/37.

While "iterable", this complete restructure of the original APIs & Clients page is a good stopping point for feedback.

## Quick glance

Screen Shot 2023-02-23 at 11 36 28 AM

You'll notice this card structure mirrors work in [other Camunda documentation](https://docs.camunda.io/docs/next/self-managed/platform-deployment/helm-kubernetes/platforms/) as well.

## Required change(s) before merge

~~- @pepopowitz I'm not sure how to reference the Optimize content here. I played around with it a bit and left my work in `code` on line 87.~~ This is technically resolved, but we'll need to figure out how to add a description/edit the styling of the cards in the next iteration.
~~- Resolve misc broken links for clean build.~~ This is temporarily fixed, but only by directly linking to `/next/` within the code. After discussing with @pepopowitz, we will likely need to override the DocCardList/DocCard components to make these cards more versatile in the longrun.

## Additional change(s) for consideration

- You'll note the descriptions in the cards reference the page description itself as outlined by [Docusaurus](https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-theme-classic/src/theme/DocCard/index.tsx). While you can hover over the card to read the full description, might it be best to shorten these for aesthetics?
- @pepopowitz and I discussed creating a new icon (rather than Docusaurus' default of a document icon,) such as a GitHub link. Thoughts here? To do this, we would need to override the Docusaurus icon and apply some swizzling.
- Shall we also rename what is in the top navigation?

## PR Checklist

- [x] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Review by akeller — COMMENTED** [review]
@christinaausley I love the direction you are doing here! The big block box buttons are great eye candy.

🔧 The Optimize box did not render for me. Looking at the code, it looks like it needs a little help to be implemented like the others in the array. 

🤔 Looking at the Public API box reminded me of the feedback on [this PR](https://github.com/camunda/camunda-platform-docs/pull/1581). In your restructure, would you consider ditching the Public API page? Even if you keep it, it feels a bit out of place when all the other boxes are component APIs.

🤔 Since Web Modeler API is still beta, would you consider a different color or some other kind of visual to make it stand out a bit from the others?

Something to explore in future iterations is the ordering on the page, floating the boxes for Clients and APIs to the top so they are "above the fold" and send the user to immediate action (or in this case, likely to immediately what they are looking for). 

Future small(ish) wins - 
* Are the icons customizable? Look at our Brand assets and pull the icons for each component API (like the [diagram](https://camunda.com/platform/) on the Platform page).
* For the text under the title, is that just the first sentence of the linked doc? Could you shorten the ones that run over, but just for this view, not the sentence in the doc itself.

Future bigger win - 
* An architecture diagram that highlights what data or functionality is available with the various APIs. This may help people better understand the components if they aren't familiar with the name.

Feel free to turn those wins into issues, or as the DRI for this project, leave me for now 😄

**Comment by pepopowitz** [comment]
> I'm not sure how to reference the Optimize content here. I played around with it a bit and left my work in code on line 87.

awwww, shoot. It's because it crosses docs instances. There _is_ no docId for the Optimize page, at least not within the rendering page's context.

We could remove the `docId` property for that item, but it would prevent the description from showing up for that item (and I think a different icon). That item would definitely stand out.

And maybe that's fine for this iteration, and we want to look more into customizing the DocCard component to be a little more versatile in a future step?

The only other alternative I can think of is that I remember reading at some point about special docusaurus content types that are there as placeholders...I'll see if I can track that down, and maybe we can create one for the Optimize page, so it can show up like the others.

File: docs/apis-clients/go-client/index.md
Comment (pepopowitz) [inline docs/apis-clients/go-client/index.md:5]: Note: This feedback is non-blocking! We could totally iterate on this.

I think what we've done for many of these descriptions is extracted the first sentence or two from the page, but I think we should put more marketing/SEO-type descriptions in each of these pages. As far as I'm aware, this field is used on two places: 1. the `meta name="description"` tag on the page, which is used by search engines to identify content, and 2. The DocCards in this PR, to show the card description. In both cases, I think the description should capture the essence of the page, which I think some of them don't.

This specific one, for example, seems like it describes a specific step in a tutorial. That's not really what the page is about -- it describes a client for interacting programmatically with Camunda, written in Go.

I'm not going to comment on each individual description, but I agree with your point in the body of this PR about making them shorter, so that more meaningful information shows up in the cards....and in doing that, I think they should also be distilled to describe why a person might come to this page, rather than be a snippet of the content on this page.

**Review by pepopowitz — DISMISSED** [review]
This looks great! Nice work on the cards, and the reorganization of info.

My only comment is non-blocking. I do think it's something we should address at some point in this work, but there's no reason for it to block this PR.

**Comment by akeller** [comment]
An additional something I thought about - if you want to change the nav bar from "APIs & Clients" to "APIs & Tools," you'll need to propagate that change across all versions of the docs (at least that's what we've had to do previously). @pepopowitz might have a more creative solution to avoid changing links and adding redirects.

**Comment by pepopowitz** [comment]
> An additional something I thought about - if you want to change the nav bar from "APIs & Clients" to "APIs & Tools," you'll need to propagate that change across all versions of the docs (at least that's what we've had to do previously). @pepopowitz might have a more creative solution to avoid changing links and adding redirects.

Do you mean we'll need to change the nav group title in [the sidebars files](https://github.com/camunda/camunda-platform-docs/blob/b98c419580a995141b6ab48af1117ccc5d74ce4a/sidebars.js#L511), or is there a built-in assumption that we would also be changing the name of the folder in source from `apis-clients` to `apis-tools`? (The second is way bigger than the first, and I want to make sure I understand the scope properly.)

**Comment by christinaausley** [comment]
@jwulf @pepopowitz I think it's also important to consider Amara's note on whether or not we should keep Public API...how do you feel about this? Are there any other APIs/tools to consider for removal?

**Comment by christinaausley** [comment]
Completed the required adjustments before merge, but will revisit some of the additional considerations and make any possible immediate changes on Monday 👍

File: docs/apis-clients/working-with-apis-clients.md
Comment (MaxTru) [inline docs/apis-clients/working-with-apis-clients.md:77]: Why does it state "Console API **clients** (REST)"?

File: docs/apis-clients/working-with-apis-clients.md
Comment (MaxTru) [inline docs/apis-clients/working-with-apis-clients.md:52]: Here (but nowhere else) we talk about SDKs. Is this on purpose?

File: docs/apis-clients/working-with-apis-clients.md
Comment (MaxTru) [inline docs/apis-clients/working-with-apis-clients.md:10]: Just an idea:

* The page has now two main concepts: (1) Deploy processes (...) using Zeebe Clients (2) Learn about Camunda Components and their APIs
* We could try to make 2 main concepts more outcome driven. For example:

1. Control your Camunda 8 process automation (e.g., deploy processes, start process instances, ...) using Zeebe Clients
2. Interact with the Camunda 8 ecosystem (e.g., retrieve open User Tasks from Tasklist) using the respective component APIs

**Review by MaxTru — DISMISSED** [review]
Cool stuff!

* For the tiles: there is sufficient space, but still we cut off the description. Could we not use more space and ideally show the entire description?
Bildschirmfoto 2023-02-27 um 09 45 18

**Comment by christinaausley** [comment]
Thanks @akeller, @pepopowitz, and @MaxTru for your helpful feedback. I've made a series of commits with the following updates:
- I have removed Public API from this landing page given the feedback on [this PR](https://github.com/camunda/camunda-platform-docs/pull/1581). Can we remove this page entirely from docs? If so, I will likely launch a separate PR for that. CC @pihme 
- As far as adjusting the styling of the DocCards, we will need to iterate moving forward to **make the cards larger so descriptions can fit (and be updated,) as well as add a border to Web Modeler API while it is in beta**, perhaps something along the lines of:
`.beta {
  border-style: solid;
  border-color: #14d890;
  border-width: 1px;
}`
I'll plan to work with @pepopowitz based on his capacity/availability to assist with the [Swizzling](https://docusaurus.io/docs/swizzling) required here. This comment also applies to **incorporating a separate icon for each API/tool**.
- I also plan to work on an architecture diagram that could more clearly demonstrate the use cases/interactions among the APIs/tools so we don't feel like we need to cover all the details of the *why* when using an API/tool in just a short description.
- I'm working through redirects here as I've only adjusted the page name and id of "working with apis and clients" to "working with apis and tools" -- @pepopowitz would you mind taking a peek at what I've been experimenting with among the redirects? I think I'm missing something and could use a fresh set of eyes to get this building properly. If you don't mind, I may also pop a bit of time on your calendar to talk through renaming the folder in its entirety 😅 I would lead this change, but wanted to get your thoughts before jumping in and scrambling things up.

File: docs/apis-clients/working-with-apis-clients.md
Comment (christinaausley) [reply docs/apis-clients/working-with-apis-clients.md]: Nice catch. Adjusted in commit 👍

File: docs/apis-clients/working-with-apis-clients.md
Comment (christinaausley) [reply docs/apis-clients/working-with-apis-clients.md]: Adjusted in commit 👍

File: docs/apis-clients/working-with-apis-clients.md
Comment (christinaausley) [reply docs/apis-clients/working-with-apis-clients.md]: Adjusted this in commit as well to incorporate the concept of outcome-driven components here.

File: product-links.txt
Comment (christinaausley) [inline product-links.txt:12]: FYI @ultraschuppi when this is ready for merge I'll sync with you on updating this product link 👍

File: docs/apis-clients/working-with-apis-tools.md
Comment (pepopowitz) [inline docs/apis-clients/working-with-apis-tools.md:44]: One thing you and I noted when pairing, which I want to posterize here, is that we don't like the fact that the objects we pass into `<DocCardList>` include a URL that is locked to one specific version. We didn't do any investigation into whether we could use the `$docs$` and `$optimize$` helpers here to prevent that, but we should iterate on that also -- maybe as part of the work we do to customize the `<DocCard>` and `<DocCardList>` components, because that's where we'd have to do it if the helpers don't work.

**Review by pepopowitz — DISMISSED** [review]

**Comment by pepopowitz** [comment]
> I'm working through redirects here as I've only adjusted the page name and id of "working with apis and clients" to "working with apis and tools" -- @pepopowitz would you mind taking a peek at what I've been experimenting with among the redirects? I think I'm missing something and could use a fresh set of eyes to get this building properly. If you don't mind, I may also pop a bit of time on your calendar to talk through renaming the folder in its entirety 😅 I would lead this change, but wanted to get your thoughts before jumping in and scrambling things up.

@christinaausley I can't see anything obvious about why the link check is failing. Can you throw a half hour on our calendars to investigate this together, and also talk about renaming the folder entirely?

File: static/.htaccess
Comment (pepopowitz) [inline static/.htaccess:17]: Oh! Not sure if you also noticed, but the remaining rule is incorrect on the right side -- it should go to `/docs/next/apis-clients...`, not `/docs/apis-clients...`, because that new URL doesn't exist yet in the current version 😅

File: src/pages/index.js
Comment (pepopowitz) [inline src/pages/index.js:33]: @christinaausley I was going to make a comment here about how we realized we can't change the URL in this location yet, because it has the host hardcoded and the new `working-with-apis-tools` URL doesn't exist yet in production....but then I also noticed that this URL is for v8.1 anyway, and this PR doesn't include any v8.1 changes.

I'm not sure if that means you want to include the title change in this PR, or if you want to wait until all of this makes it into v8.1? I have no preference, but since you pointed out my misunderstanding of the versions impacted by this PR, I realized I was interpreting this specific change incorrectly all along.

File: static/.htaccess
Comment (christinaausley) [reply static/.htaccess:17]: THANK YOU! 🥇

**Comment by christinaausley** [comment]
This build is running clean and the first iteration is ready for final review.

- @pepopowitz Note that I did indeed have to keep the product name in its original form to get the build to complete 👍 A good note to keep in mind as we're adjusting product links moving forward.

- @pepopowitz and I have a meeting set for next week to accomplish two things (of which will both be tackled in separate PRs):
1. Talk through change of APIs & Clients nav tab to APIs & Tools so I know exactly what to prepare for when we make this change.
2. Talk through Swizzling so I can get a better understanding of the process in launching a second iteration of this page, ideally which incorporates discussed components such as expanding the boxes to include full descriptions, adding a border color to Beta, changing the icons, and more.

**Review by pepopowitz — APPROVED** [review]
This is awesome, thanks for your persistence on this!

**Review by akeller — APPROVED** [review]
:shipit: all these changes are on `next`, so feel free to announce it on the #ask-documentation channel for more feedback when it's live!

**Pull request merged**
